elide-spring/elide-spring-boot-autoconfigure/src/main/java/com/yahoo/elide/spring/config/ElideDynamicConfiguration.java

```diff
@@ -0,0 +1,148 @@
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.orm.jpa.LocalContainerEntityManagerFactoryBean;
+import org.springframework.orm.jpa.persistenceunit.PersistenceUnitManager;
+import org.springframework.orm.jpa.vendor.HibernateJpaVendorAdapter;
+
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+import javax.persistence.Entity;
+import javax.persistence.spi.PersistenceUnitInfo;
+import javax.sql.DataSource;
+
+/**
+ * Dynamic Configuration For Elide Services. Override any of the beans (by
+ * defining your own) and setting flags to disable in properties to change the
+ * default behavior.
+ */
+
+@Slf4j
+@Configuration
+@EnableConfigurationProperties(ElideConfigProperties.class)
+@ConditionalOnExpression("${elide.dynamic-config.enabled:false}")
+public class ElideDynamicConfiguration {
+
+    public static final String HIBERNATE_DDL_AUTO = "hibernate.hbm2ddl.auto";
+
+    /**
+     * Configure factory bean to create EntityManagerFactory for Dynamic Configuration.
+     * @param source :DataSource for JPA
+     * @param jpaProperties : JPA Config Properties
+     * @param hibernateProperties : Hibernate Config Properties
+     * @param dynamicCompiler : ElideDynamicEntityCompiler
+     * @return LocalContainerEntityManagerFactoryBean bean
+     */
+    @Bean
+    public LocalContainerEntityManagerFactoryBean entityManagerFactory (
+            DataSource source,
+            JpaProperties jpaProperties,
+            HibernateProperties hibernateProperties,
+            ObjectProvider<ElideDynamicEntityCompiler> dynamicCompiler) {
+
+        try {
+
+            //Map for Persistent Unit properties
+            Map<String, Object> puiPropertyMap = new HashMap<>();
+
+            //Bind entity classes from classpath to Persistence Unit
+            ArrayList<Class> bindClasses = new ArrayList<>();
+            bindClasses.addAll(ClassScanner.getAnnotatedClasses(Entity.class));
+
+            //Bind FromTable/FromSubSelect classes from classpath to Persistence Unit
+            bindClasses.addAll(ClassScanner.getAnnotatedClasses(FromTable.class));
+            bindClasses.addAll(ClassScanner.getAnnotatedClasses(FromSubquery.class));
+
+            //Map of JPA Properties to be be passed to EntityManager
+            Map<String, String> jpaPropMap = jpaProperties.getProperties();
+            String hibernateGetDDLAuto = hibernateProperties.getDdlAuto();
+
+            //Set the relevant property in JPA corresponding to Hibernate Property Value
+            if (jpaPropMap.get(HIBERNATE_DDL_AUTO) == null && hibernateGetDDLAuto != null) {
+               jpaPropMap.put(HIBERNATE_DDL_AUTO, hibernateGetDDLAuto);
```

Comment:
There are a lot of hibernate properties we will likely want to preserve beyond just the DDL_AUTO - batch settings, fetch modes, connection pool settings, etc.  Can we just take the entire map of hibernate properties and use them all?

Reply:
most of the hibernate properties are passed as `spring.jpa.properties.hibernate.*`. But there are 4 properties which are passed as `spring.jpa.hibernate.*` . Here we are taking those 4 properties and setting the corresponding parameters in `spring.jpa.properties.hibernate.*` if not already set so that it gets passed along to the EntityManagerFactoryBean. https://docs.spring.io/spring-boot/docs/current/reference/html/appendix-application-properties.html

Reply:
Ayes has expanded the logic to remaining 3 params too.